### PR TITLE
Start, Stop and Delete API 

### PR DESCRIPTION
**Pull request by flyjwayur**
Is it better to make the separate API for start and stop ?:) or both can be implemented as one API?
In main.js, variable app needs name property or not? 
Why renderAppDetail(app.id) is not necessary to be written here?
Which part can be implemented as API in indexAPI.js and which part can be remain in main.js? 
Currently, I took the change part and put it into the indexAPI.js as 'applist.start API'

``` js
$startStopAppButton.on('click', function () {
  var app = {
    id: $('#name-app').data('id')
  }
  applist.start(app)
  .then(function(app){
    $startStopAppButton.find('i').toggleClass('glyphicon-play glyphicon-stop')
    $startStopAppButton.toggleClass('main-button')
    // change
    var label = $startStopAppButton.find('span').text().trim() === 'Start' ? 'Stop' : 'Start'
    $startStopAppButton.find('span').text(label)
    //renderAppDetail(app.id)
  })
})
```

### CODE REVIEW

File: scripts/indexApi.js [hunk 1]
```diff
@@ -10,41 +10,41 @@ applist.findAll = function () {
...
-        } else {
-          return currentApp
-        }
-      })
-      localStorage.setItem('apps', JSON.stringify(newApps))
-      return app
+  .then(function (apps) {
+    var newApps =apps.map(function (currentApp) {
+      if (currentApp.id === app.id) {
+        return {name:app.name, id:app.id}
```
Comment (flyjwayur) [inline scripts/indexApi.js]: state of an app (start and stop)

File: scripts/indexApi.js [hunk 2]
```diff
@@ -10,41 +10,41 @@ applist.findAll = function () {
...
+      } else {
+        return currentApp
+      }
     })
+    localStorage.setItem('apps', JSON.stringify(newApps))
+    return app
+  })
+}
+
+applist.start = function(app){
```
Comment (flyjwayur) [inline scripts/indexApi.js:49]: Use applist.update
ex app.update === start

**Comment by gr2m** [comment]
@titay2 I found out what the problem was with the remove, I was surprised myself! 

When we set `data-id` with `.attr` and then use `.data` to access it, it will return the value we set with `.attr`. But if we set it to another value with `.attr` then `.data` will still return the _first_ value. I wonder if this is actually a jQuery bug.

Here is an example of what I mean

``` js
$body = $(document.body)
$body.attr('data-id', '1')
console.log($body.data('id'))
// 1
$body.attr('data-id', '2')
console.log($body.data('id'))
// still 1
```

Here are a few changes that I suggest that fix the problem

``` dif
diff --git a/scripts/main.js b/scripts/main.js
index b0dde97..2de0606 100644
--- a/scripts/main.js
+++ b/scripts/main.js
@@ -58,7 +58,6 @@ $updateAppForm.on('submit', function (event) {
     id: $('#name-app').data('id'),
     name: changed
   }
-  console.log("app:"+ JSON.stringify(app))
   if (changed) {
     applist.update(app)
     .then(function (app) {
@@ -71,17 +70,14 @@ $updateAppForm.on('submit', function (event) {

 $deleteButton.on('click',function(event){
   event.preventDefault()
-  var app = {
-    id: $('#delete-button').data('id')
-  }
+  var id = $('#name-app').data('id')
   //var id = $('#delete-button').data('id')
   //console.log(id)
-  console.log(id)
-  applist.find(id)
+  console.log('#name-app data-id', id)
+  debugger
+  applist.remove(id)
   .then (function(app){
-    applist.remove(app)
     setRoute('')
-    console.log("applist.remove(id)"+applist.remove(app))
   })
 })

@@ -148,8 +144,7 @@ function renderAppDetail (id) {

   .then(function (app) {
     $('#name-app').html(app.name)
-    $('#name-app').attr('data-id', app.id)
-    $('#delete-Button').attr('data-id', app.id)
+    $('#name-app').data('id', app.id)
     $('#detail-app-container').attr('data-state',app.state)
     $('#folder').html('~Hoodie/' + app.name)
     $body.attr('data-state', 'app-detail')
```

**Comment by flyjwayur** [comment]
@gr2m @titay2 Hey :D After our discussion, the selector for the variable id is changed from 'name-app' to 'detail-app-container' for the deleteButton.

**Pull request merged**